### PR TITLE
tests: tests from others and kubevirt

### PR DESCRIPTION
**Pull request by mintbomb27**
## Related Issue(s)
#546 

## Description
Contains 10 new kuttl tests from others category in addition to `enforce-instancetype` from kubevirt.

## Checklist

- [x] I have read the [policy contribution guidelines](https://github.com/kyverno/policies/blob/main/README.md#contribution).
- [x] I have added test manifests and resources covering both positive and negative tests that prove this policy works as intended.
- [ ] I have added the artifacthub-pkg.yml file and have verified it is complete and correct.

### CODE REVIEW

**Review by chipzoller — COMMENTED** [review]
Looks good, just one thing.

File: other/add-certificates-volume/pod-resources.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  annotations:
+    inject-certs: enabled
```
[the remark this answers is not in the document]
Comment (mintbomb27) [reply other/add-certificates-volume/pod-resources.yaml:5]:
```sh
case.go:369: admission webhook "mutate.kyverno.svc-fail" denied the request: mutation policy add-certificates-volume error: failed to apply policy add-certificates-volume rules [add-ssl-certs: failed to evaluate preconditions: failed to substitute variables in condition key: failed to resolve request.object.metadata.annotations."inject-certs" at path : JMESPath query failed: Unknown key "annotations" in path]```

It breaks! Can't believe I didn't try that out.

[the remark this answers is not in the document]
Comment (mintbomb27) [reply other/add-certificates-volume/pod-resources.yaml:5]: Both, if it doesn't have an `inject-certs` annotation, or if it doesn't have any annotations at all, it breaks.

[the remark this answers is not in the document]
Comment (chipzoller) [reply other/add-certificates-volume/pod-resources.yaml:5]: Right, that's as I suspected. We need to add an [existence check](https://kyverno.io/docs/writing-policies/jmespath/#non-existence-checks) to the policy. You then need to test for that with one or two more test resources.

[the remark this answers is not in the document]
Comment (mintbomb27) [reply other/add-certificates-volume/pod-resources.yaml:5]: Fixed!

File: other/add-certificates-volume/add-certificates-volume.yaml [hunk 2]
```diff
@@ -26,7 +26,7 @@ spec:
           - Pod
     preconditions:
       all:
-      - key: '{{request.object.metadata.annotations."inject-certs"}}'
+      - key: '{{request.object.metadata.annotations."inject-certs" || " "}}'
```
Comment (chipzoller) [inline other/add-certificates-volume/add-certificates-volume.yaml:29]: Empty string, please, not space.

**Review by chipzoller — CHANGES_REQUESTED** [review]
Thanks for adding this. Just need to use empty string (`''` or `""` in this case) rather than space.

File: other/add-certificates-volume/add-certificates-volume.yaml (hunk 2 above)
Comment (mintbomb27) [reply other/add-certificates-volume/add-certificates-volume.yaml]: Fixed.

**Review by chipzoller — APPROVED** [review]
Thanks, looks good.

**Pull request merged**